radix-engine-common/src/math/decimal.rs

```diff
@@ -740,36 +740,51 @@ impl FromStr for Decimal {
     type Err = ParseDecimalError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let tens = I192::TEN;
         let v: Vec<&str> = s.split('.').collect();
 
-        let mut int = match I192::from_str(v[0]) {
+        let integer_part = match I192::from_str(v[0]) {
             Ok(val) => val,
-            Err(_) => return Err(ParseDecimalError::InvalidDigit),
+            Err(err) => match err {
+                ParseI192Error::NegativeToUnsigned => unreachable!("Not possible to be thrown by from_str"),
+                ParseI192Error::Overflow => return Err(ParseDecimalError::Overflow),
+                ParseI192Error::InvalidLength => unreachable!("Not possible to be thrown by from_str"),
+                ParseI192Error::InvalidDigit => return Err(ParseDecimalError::InvalidDigit),
+                ParseI192Error::Empty => I192::ZERO,
```

Comment:
What is `Empty`? Same for all the other places.

Reply:
As discussed. This is to support `.123` and `123.`

Reply:
Although we have now agreed to not allow this. Have tweaked it and added a comment.